lib/functions/transferpartsjoin.js

```diff
@@ -63,6 +63,9 @@ class JoinTransferParts extends AsyncGeneratorFunction {
                     completedRanges: new DRange()
                 };
                 this.trackedAssets.set(sourceUrl, trackedAsset);
+                this.controller.emit(TransferEvents.TRANSFER_START, {
+                    transferAsset
+                });
```

Comment:
The reason I removed this was because this is _after_ the first block has been transferred. Might be better to send this out during `CreateTransferParts` after the part has been yielded.